t/codeclimate_spec.vim

```diff
@@ -16,3 +16,43 @@ describe "codeclimate#ParseIssues"
     Expect l:parsedIssues == l:expectedIssues
   end
 end
+
+function! MockRunAnalysis()
+  let s = 'function! ' . codeclimate#sid() . 'RunAnalysis(args)' . "\n"
+  let s = s . '  let g:t_runopts = a:args' . "\n"
+  let s = s . 'endfunction'
+  execute s
+  let g:t_runopts = "not-called"
+endfunction
+
+function! EscapePath(path)
+  " This is pretty wacky, and I'm not sure why the strings get so mangled.  CLI
+  " invocations seem fine when actually used & echoed for debugging, so I think
+  " this is some weird vspec mangling, and this escaping works around it.
+  " ¯\_(ツ)_/¯
+  return '''''\''''' . a:path . '''\'''''''
+endfunction
+
+
+describe "buffer opts"
+  before
+    call MockRunAnalysis()
+  end
+
+  describe "CodeClimateAnalyzeCurrentFile"
+    it "handles no buffer var set"
+      silent edit 'fixtures/foo.rb'
+      call Call("s:AnalyzeCurrentFile")
+      let l:expected = ' ' . EscapePath('fixtures/foo.rb')
+      Expect g:t_runopts ==# l:expected
+    end
+
+    it "handles buffer var"
+      silent edit 'fixtures/foo.rb'
+      let b:codeclimateflags = "--engine rubocop"
+      call Call("s:AnalyzeCurrentFile")
+      let l:expected = '--engine rubocop ' . EscapePath('fixtures/foo.rb')
+      Expect g:t_runopts ==# l:expected
```

Comment:
Sorta. `'` does no escaping.

I used `"` for the `let` since that feels more like what somebody would type in their `vimrc`, while the `'` for the expected values is a nice way of keeping the actual expected values very predictable.

Changing the lets to `'` for the tests would work fine, though.